sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -149,6 +149,9 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       case AlterTable(_, _, u: UnresolvedV2Relation, _) =>
         failAnalysis(s"Table not found: ${u.originalNameParts.quoted}")
 
+      case DescribeColumn(_: ResolvedTable, UnresolvedAttribute(colNameParts), _) =>
+        failAnalysis(s"Column not found: ${colNameParts.quoted}")
```

Comment:
This check should come before expression checks. Otherwise, we will get `org.apache.spark.sql.AnalysisException: cannot resolve '`invalid_col`' given input columns: []` and this is misleading since input columns are printed as empty.